sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2877,4 +2877,28 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
   override def visitRefreshTable(ctx: RefreshTableContext): LogicalPlan = withOrigin(ctx) {
     RefreshTableStatement(visitMultipartIdentifier(ctx.multipartIdentifier()))
   }
+
+  /**
+   * A command for users to list the column names for a table.
+   * This function creates a [[ShowColumnsStatement]] logical plan.
+   *
+   * The syntax of using this command in SQL is:
+   * {{{
+   *   SHOW COLUMNS (FROM | IN) tableName=multipartIdentifier
+   *        ((FROM | IN) namespace=multipartIdentifier)?
+   * }}}
+   */
+  override def visitShowColumns(ctx: ShowColumnsContext): LogicalPlan = withOrigin(ctx) {
+    val table = visitMultipartIdentifier(ctx.tableName)
+    if (ctx.namespace != null) {
+      val namespace = visitMultipartIdentifier(ctx.namespace)
+      if (table.length > 1 && namespace.length > 1) {
+        throw new ParseException(
+          s"If namespace is defined, table can not be multipart: ${table.mkString(".")}", ctx)
```

Comment:
let's use `quoted` instead of mkString since it's multi part identifier.

Reply:
Also, the exception message is different from what is actually checked (defined AND namespace.length > 1).

Reply:
@imback82 
I have change the message with this:
"_If namespace is multipart, table can not be multipart:_"
 but i'm not sure what are you referring with quoted instead mkstring.

Reply:
You can use `MultipartIdentifierHelper.quoted`. Check `visitCacheTable` for an example.

Reply:
Ok, I have seen it, thank you